matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/room/location/LocationSharingService.kt

```diff
@@ -59,5 +61,13 @@ interface LocationSharingService {
     /**
      * Returns a LiveData on the list of current running live location shares.
      */
+    @MainThread
```

Comment:
can these only be called by the main thread? I was under the impression `LiveData's` observing mechanism would handle that for us

Reply:
In fact, I added this annotation because the implementation relies on `Monarchy.findAllMappedWithChanges` method. In this method, there is an `assertMainThread()` to ensure we are in the main thread. If it is not the case, it throws an exception.
Ideally, we should not rely on `LiveData` but only on `Flow`. However, last time I tried using the `toFlow` extension available in `Realm`, it has lead to problems about Frozen objects...

Reply:
I see, thanks for the explanation :+1: